src/pasttle/server.py

```diff
@@ -327,6 +327,64 @@ def showdiff(db, parent, id):
     )
 
 
+@bottle.get('/delete/<id:int>')
+def deletepasteconfirm(db, id):
+    """ Shows a confirmation form before letting you delete your paste."""
+
+    return template(
+        'delete',
+        url=get_url(),
+        title=util.conf.get(util.cfg_section, 'title'),
+        version=pasttle.__version__,
+    )
+
+
+@bottle.post('/delete/<id:int>')
+def deletepaste(db, id):
+    """ Clears the contents of a paste. """
+
+    def _delete_and_return(paste):
+        db.delete(paste)
+        db.commit()
+
+        return bottle.redirect('{0}/{1}'.format(get_url(), paste.id, ))
+
+    paste = _get_paste(db, id)
+    form = bottle.request.forms
+    if form.get('confirm') and form.get('confirm') == 'No':
```

Comment:
I was going to say "I'm not sure we can do that since we're actually potentially handling two separate form posts, but you're totally right -- that's definitely better.